Build failure: nheko

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Build *nheko*

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310615881

### Relevant log output

```console
error: Cannot build '/nix/store/gjdrd7381gmhc3a15vndxqzp4305f20i-nheko-0.12.1.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/hvgdcjsm08nmxx3hr3m9a90l3gpqbkr2-nheko-0.12.1
       Last 25 log lines:
       >
       > -- Configuring done (6.0s)
       > CMake Error at CMakeLists.txt:909 (target_link_libraries):
       >   Target "nheko" links to:
       >
       >     Qt::QmlPrivate
       >
       >   but the target was not found.  Possible reasons include:
       >
       >     * There is a typo in the target name.
       >     * A find_package call is missing for an IMPORTED target.
       >     * An ALIAS target is missing.
       >
       >
       >
       > -- Generating done (0.1s)
       > CMake Warning:
       >   Manually-specified variables were not used by the project:
       >
       >     BUILD_TESTING
       >     CMAKE_EXPORT_NO_PACKAGE_REGISTRY
       >     CMAKE_POLICY_DEFAULT_CMP0025
       >
       >
       > CMake Generate step failed.  Build files cannot be regenerated correctly.
       For full logs, run:
         nix log /nix/store/gjdrd7381gmhc3a15vndxqzp4305f20i-nheko-0.12.1.drv
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.4, NixOS, 25.11 (Xantusia), 25.11.20251019.5e2a59a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.32.1`
 - nixpkgs: `/nix/store/7agp54mgffm9m1wc1kgmkm37pvy18qhf-source`

### Notify maintainers

@Ekleog @fpletz @rebmit @rnhmjoj

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/nh/nheko/package.nix
+++ b/pkgs/by-name/nh/nheko/package.nix
@@ -43,6 +43,11 @@ stdenv.mkDerivation (finalAttrs: {
       hash = "sha256-y8aiS6h5CSJYBdsAH4jYhAyrFug7aH2H8L6rBfULnQQ=";
     })
     ./fix-darwin-build.patch
+    # Fix for Qt 6.10
+    (fetchpatch {
+      url = "https://github.com/Nheko-Reborn/nheko/commit/af2ca72030deb14a920a888e807dc732d93e3714.patch";
+      hash = "sha256-tlYrfEoUkdJoVzvfF34IhXdn1AxLO0MOlp9rzuFivws=";
+    })
   ];
 
   nativeBuildInputs = [

